lib/auth/auth_with_roles.go

```diff
@@ -5532,6 +5532,10 @@ func (a *ServerWithRoles) CreateDatabase(ctx context.Context, database types.Dat
 	if err := a.checkAccessToDatabase(database); err != nil {
 		return trace.Wrap(err)
 	}
+	// Don't allow discovery service to create databases with dynamic labels.
+	if a.hasBuiltinRole(types.RoleDiscovery) && len(database.GetDynamicLabels()) > 0 {
```

Comment:
Can you please include the same check into kube endpoints?

Reply:
> Can you please include the same check into kube endpoints?

Thanks! I missed as somehow i though kube cluster doesn't have dynamic labels. Added in [98d53a7](https://github.com/gravitational/teleport/pull/29717/commits/98d53a73ea028ba4745dc8f09cac0cbc209fd730)

> What is identity was generated for DBDiscovery and DBDatabse so the a.hasBuiltinRole(types.RoleDiscovery) && a.hasBuiltinRole(types.RoleDatabase) == true and CreateDatabase method was called from db agent process ?

> DB service doesn't call CreateDatabase right? Worth testing this scenario but I don't think it should be a problem

Database service doesn't call `CreateDatabase`. Also, even if it does it wouldn't be a problem:
https://github.com/gravitational/teleport/blob/8a1d0e23c85154b253517a6f5ed782c9dd0e1d55/lib/service/db.go#L42-L45

Different service in the same process with use its own auth connector with a single system role.